src/core/datatypes/trees/Tree.cpp

```diff
@@ -2052,10 +2051,24 @@ TopologyNode& Tree::reverseParentChild(TopologyNode &n)
         TopologyNode &p = n.getParent();
         ret = &(reverseParentChild(p));
 
+        // save p values
+        size_t idx_p = p.getIndex();
+        double brlen_p = p.getBranchLength();
+        std::vector<std::string> ncomm_p = p.node_comments;
+        std::vector<std::string> bcomm_p = p.branch_comments;
```

Comment:
I think we can consolidate three lines into one if we swap branch comments using:
```
std::swap(p.branch_comments, n.branch_comments);
```

Normally swapping a and b involves:
```
tmp = a;
a = b;
b = tmp;
```

std::swap does all this at once, plus is uses "move construction" to avoid actually copying object contents with expensive objects like std::vector.